codecs/libavcodec.rs

```diff
@@ -472,13 +474,13 @@ impl videodecoder::VideoDecoder for VideoDecoderImpl {
     fn decode_frame(&self, data: &[u8], presentation_time: &Timestamp)
                     -> Result<Box<videodecoder::DecodedVideoFrame + 'static>,()> {
         let mut data: Vec<_> = data.iter().map(|x| *x).collect();
-        for _ in range(0, FF_INPUT_BUFFER_PADDING_SIZE) {
+        for _ in 0..FF_INPUT_BUFFER_PADDING_SIZE {
             data.push(0);
         }
 
         let mut packet = AvPacket::new(data.as_mut_slice());
-        let presentation_time = *presentation_time;
-        self.context.borrow_mut().set_get_buffer_callback(Box::new(|frame: &AvFrame| {
+        let presentation_time = presentation_time.clone();
```

Comment:
I'm getting a lifetime error when I remove that line. I also get it when I write `let presentation_time = presentation_time`. clone works.